src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/NotebookQueueState.java

```diff
@@ -470,13 +480,14 @@ private NotebookExecRange getNotebookExecRange(Scope scope, Range range)
            row++)
       {
          String line = docDisplay_.getLine(row);
-         for (int col = 0; col <= line.length(); col++)
+         int lineLength = codePointLength(line);
+         for (int col = 0; col <= lineLength; col++)
          {
             if (startPos.getRow() == row && startPos.getColumn() == col)
             {
                start = pos;
             }
-            else if (endPos.getRow() == row && endPos.getColumn() == col)
+            else if (endPos.getRow() == row && (endPos.getColumn() == col || col == lineLength))
```

Comment:
Why is the `col == lineLength` check needed? (Maybe add a comment explaining?)

Reply:
I can add a comment here. It's to correctly count the end position since the `endPos` may be greater than the char array length.